### ISSUE TITLE
NumLock does not work properly in WX mode

### ISSUE THREAD

**Issue by alexandrustinov** [elfmz/far2l#1494]
How to reproduce: 

Open Text Editor, activate the NumLock, then try to type numbers using the digital keyboard at right side of keyboard.

Expected behavior: Far2L should add text with numbers
Actual behavior: Far2L tries to navigate, despite NumLock state

The same happens in Command Line, Shift+F4 dialog and so on

**Comment by elfmz** [comment]
Whats OS? For me in Ubuntu 18 it works correctly.

**Comment by alexandrustinov** [comment]
CentOS Stream 9

```
ldd ./far2l_gui.so
        linux-vdso.so.1 (0x00007ffd00292000)
        libwx_gtk3u_core-3.0.so.0 => /lib64/libwx_gtk3u_core-3.0.so.0 (0x00007f84a1200000)
        libwx_baseu-3.0.so.0 => /lib64/libwx_baseu-3.0.so.0 (0x00007f84a0e00000)
```
points to  wxGTK3-3.0.5.1-6.el9.x86_64.rpm

here the wx info

```
wxBase.x86_64                                        3.2.1-3.el9                        @epel
wxBase-devel.x86_64                                  3.2.1-3.el9                        @epel
wxBase3.x86_64                                       3.0.5.1-6.el9                      @epel
wxBase3-devel.x86_64                                 3.0.5.1-6.el9                      @epel
wxGTK.x86_64                                         3.2.1-3.el9                        @epel
wxGTK-devel.x86_64                                   3.2.1-3.el9                        @epel
wxGTK-gl.x86_64                                      3.2.1-3.el9                        @epel
wxGTK-i18n.noarch                                    3.2.1-3.el9                        @epel
wxGTK-media.x86_64                                   3.2.1-3.el9                        @epel
wxGTK-webview.x86_64                                 3.2.1-3.el9                        @epel
wxGTK3.x86_64                                        3.0.5.1-6.el9                      @epel
wxGTK3-devel.x86_64                                  3.0.5.1-6.el9                      @epel
wxGTK3-gl.x86_64                                     3.0.5.1-6.el9                      @epel
wxGTK3-i18n.noarch                                   3.0.5.1-6.el9                      @epel
wxGTK3-media.x86_64                                  3.0.5.1-6.el9                      @epel
wxGTK3-webview.x86_64                                3.0.5.1-6.el9                      @epel
```

**Comment by elfmz** [comment]
wayland again. looks like this bug - https://github.com/wxWidgets/wxWidgets/pull/2255

**Comment by alexandrustinov** [comment]
I just checked filezilla, it uses the same wx package - it seems NumLock works properly there.
Also I rebuilt against wx3.2 from list above (just deleting 3.0.5.1 and rebuilding once again) - the same issue persist, while WX's .so were changed.

**Comment by elfmz** [comment]
Checked wx sources, they still dont support NUMLOCK query - https://github.com/wxWidgets/wxWidgets/blob/master/src/unix/utilsx11.cpp#L2603
although it must be possible since GTK3 by using gdk_keymap_get_num_lock_state...

**Comment by alexandrustinov** [comment]
epel uses wxWidgets 3.2.1, last major release. I'd raise ticket for wxWidgets team, they should fix that in any case.

I also see next callls in code
		`if (wxGetKeyState(WXK_SCROLL))`
		`if (wxGetKeyState(WXK_CAPITAL))`

Not sure that it's critical, it seems that CAPSLOCK_ON and SCROLLLOCK_ON are used for logging only.

**Comment by elfmz** [comment]
You can also remove g_broadway here under if: https://github.com/elfmz/far2l/blob/master/WinPort/src/Backend/WX/wxWinTranslations.cpp#L455
To make sure it not works. I already checked and saw that it doesnt for my my wayland-based system. This if exists there exactly due to wx in debug build raised errors when tried to query numlock under Wayland.

**Comment by alexandrustinov** [comment]
removed g_broadway, then g_wayland - nothing helps at the moment.

maybe it makes sense to implement temporary workaround function which will call gdk_keymap_get_num_lock_state() instead of wxGetKeyState() for linux/wayland only (to inject the bugfix).

as for all that is not so critical, I already had to patch wxWidgets to enable XF86Copy, XF86Paste, XF86Cut, XF86Undo, XF86Redo, probably it's good time to make bug report/pull request for them.

**Comment by elfmz** [comment]
i'll try and if it works will send PR to them

**Comment by elfmz** [comment]
Created issue for them: https://github.com/wxWidgets/wxWidgets/issues/23195

**Comment by elfmz** [comment]
They've accepted fix and merged it into 3.2, so now its only matter of (long) time when it will be everywhere... though need to do some adjustments in far2l's code to be ready for that ..

**Comment by alexandrustinov** [comment]
The issue is resolved with

wxBase.x86_64 3.2.2.1-1.el9 @epel-testing
